vassal-app/src/main/java/VASSAL/build/module/DoActionButton.java

```diff
@@ -116,22 +116,24 @@ public class DoActionButton extends AbstractToolbarItem
   protected boolean loopPropertyRegistered = false;
 
   public DoActionButton() {
-    final ActionListener rollAction = e -> {
-      try {
-        doActions();
-      }
-      catch (RecursionLimitException ex) {
-        RecursionLimiter.infiniteLoop(ex);
-      }
-    };
-
-    setLaunchButton(makeLaunchButton(
-      getConfigureTypeName(),
-      getConfigureTypeName(),
-      "",
-      rollAction
-    ));
-    launch = getLaunchButton(); // for compatibility
+    if (!(this instanceof SpecialDiceButton)) {
```

Comment:
Woah, shouldn't this be in the ctor of `SpecialDiceButton`?

Reply:
I agree it looks shitty but I couldn't figure out any way inside of SpecialDiceButton's ctor to prevent it calling the *empty* ctor of DoActionButton, since that seems to be what is done by default? Do you know a way?

Reply:
The ` SpecialDiceButton` ctor calls the nullary `DoActionButton` ctor first because there's no explicty call to `super(something, else)` and some ctor of the parent class must be called. You could get around this by adding another `DoActionButton` ctor for `SpecialDiceButton` to call explicitly.

Reply:
I've added a new version of the constructor with a dummy parameter so I can avoid calling the nullary one. Seems a little more obfuscatory to me, but...